src/components/CalendarPicker/index.js

```diff
@@ -0,0 +1,176 @@
+        this.onDayPress = this.onDayPress.bind(this);
+    }
+
+    // eslint-disable-next-line rulesdir/prefer-early-return
+    componentDidMount() {
+        if (this.props.minDate && this.props.maxDate && this.props.minDate > this.props.maxDate) {
+            throw new Error('Minimum date cannot be greater than the maximum date.');
+        }
+    }
+
+    // eslint-disable-next-line no-shadow, rulesdir/prefer-early-return
+    componentDidUpdate(prevProps, prevState) {
+        if (this.props.defaultYear !== prevProps.defaultYear && this.props.defaultMonth !== prevProps.defaultMonth) {
+            this.setState(prev => ({...prev, currentDateView: moment(prev.currentDateView).set('year', this.props.defaultYear).set('month', this.props.defaultMonth).toDate()}));
+        } else if (this.props.defaultYear !== prevProps.defaultYear) {
+            this.setState(prev => ({...prev, currentDateView: moment(prev.currentDateView).set('year', this.props.defaultYear).toDate()}));
+        }
+
+        if (prevState.currentDateView !== this.state.currentDateView && this.props.onChanged) {
+            this.props.onChanged(this.state.currentDateView);
+        }
+    }
+
+    /**
+     * Updates the currentDateView state by subtracting one month from it.
+     * @returns {void}
+     */
+    onPrevMonthPress() {
+        this.setState(prev => ({currentDateView: moment(prev.currentDateView).subtract(1, 'M').toDate()}));
+    }
+
+    /**
+     * Updates the currentDateView state by adding one month to it.
+     * @returns {void}
+     */
+    onNextMonthPress() {
+        this.setState(prev => ({currentDateView: moment(prev.currentDateView).add(1, 'M').toDate()}));
+    }
+
+    /**
+     * Sets the yearPickerVisible state to true, displaying the year picker component.
+     * @returns {void}
+     */
+    onYearPickerPress() {
+        const minYear = this.props.minDate ? Number(moment(this.props.minDate).year()) : 1970;
+        const maxYear = this.props.maxDate ? Number(moment(this.props.maxDate).year()) : 1970 + 200;
+        const currentYear = this.state.currentDateView.getFullYear();
+        Navigation.navigate(ROUTES.getYearSelectionRoute(minYear, maxYear, currentYear, Navigation.getActiveRoute()));
+
+        if (this.props.onYearPressed) {
+            this.props.onYearPressed();
+        }
+    }
+
+    /**
+     * Calls the onSelected function with the selected date, if it is within the min/max range.
+     * @param {number} day - The day of the month that was selected.
+     * @returns {void}
+     */
+    onDayPress(day) {
+        if (!this.props.onSelected) {
+            return;
+        }
+        const selectedDate = new Date(this.state.currentDateView.getFullYear(), this.state.currentDateView.getMonth(), day);
+        if ((this.props.minDate && moment(selectedDate) < moment(this.props.minDate).startOf('day'))
+        || (this.props.maxDate && moment(selectedDate) > moment(this.props.maxDate).startOf('day'))) {
```

Comment:
Let's split that condition into variables to make this condition more readable

```suggestion
        const isBeforeMinDate = this.props.minDate && moment(selectedDate) < moment(this.props.minDate).startOf('day');
        const isAfterMaxDate = this.props.maxDate && moment(selectedDate) > moment(this.props.maxDate).startOf('day');
        if (isBeforeMinDate || isAfterMaxDate) {
            return;
```